src/Dfe.PlanTech.AzureFunctions/Models/MappedEntity.cs

```diff
@@ -135,15 +144,13 @@ private void UpdateEntityStatusByEvent()
                     throw new CmsEventException(string.Format("Content with Id \"{0}\" has event 'unpublish' despite not existing in the database!", IncomingEntity.Id));
                 }
                 IncomingEntity.Published = false;
```

Comment:
Ah so, for the incoming entity, we set its 3 status fields to whatever existing entity is, and then based on the CmsEvent, one of those may get updated

Why do we not want to change the existing status fields? Is it because this could happen? user unpublishes and archives:
- Webhook payload 1 has unpublish = true archive = false
- Webhook payload 2 has unpublish = false archive = true (this should not happen because 2nd payload should have the knowledge of both events?)
so its not safe to update both, or there another reason

Reply:
`does incoming not have all the statuses we want though?` that's exactly it mate. An incoming payload does not contain information on the status of the content [^1]. We only[^1] know the event that _triggered_ the payload; i.e. whether it was a publish, unpublish, save, archive, unarchive, etc.

As a result it would more be like

- Payload 1 - publish event - entity is now `published`, `unarchived`, `undeleted`
- Payload 2 - save event - entity is now `unpublished`, `unarchived`, `undeleted`